sql: implement the pg_timezone_names table

otan opened the pull request:
Informs #84505

Release note (sql change): Implement the `pg_timezone_names` pg_catalog table, which lists all supported timezones.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94122)
<!-- Reviewable:end -->

rafiss reviewed (approved):
this is timely! we just got https://github.com/cockroachdb/docs/issues/15873, and looks like this will resolve it

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94122#-:-NJsX_IVEqm9ZAkziUi2:b-lqs1ja)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[`pkg/sql/pg_catalog.go` line 4154 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94122#-NJsYzQODs-wKbvFM0L8:-NJsYzQODs-wKbvFM0L9:byh06nq) ([raw file](https://github.com/cockroachdb/cockroach/blob/09c571eb5b081ae859f928cbca30beb2e0e0f099/pkg/sql/pg_catalog.go#L4154)):*
> ```Go
> func addRowForTimezoneNames(tz string, t time.Time, addRow func(...tree.Datum) error) error {
> 	abbrev, offset := t.Zone()
> 	d := duration.MakeDuration(int64(offset)*int64(time.Second), 0, 0)
> ```

what is this multiplication meant to calculate?
___
*[`pkg/sql/pg_catalog.go` line 4159 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94122#-NJsXLJG8Wo6wTiF99vb:-NJsXLJG8Wo6wTiF99vc:b7tyrkr) ([raw file](https://github.com/cockroachdb/cockroach/blob/09c571eb5b081ae859f928cbca30beb2e0e0f099/pkg/sql/pg_catalog.go#L4159)):*
> ```Go
> 		tree.NewDString(abbrev), // abbrev
> 		tree.NewDInterval(d, types.DefaultIntervalTypeMetadata), // utc_offset
> 		tree.MakeDBool(tree.DBool(t.IsDST())),                   // is_dist
> ```

nit: comment should be `is_dst`


<!-- Sent from Reviewable.io -->

otan committed:
sql: implement the pg_timezone_names table

Release note (sql change): Implement the `pg_timezone_names` pg_catalog
table, which lists all supported timezones.

otan commented:
bors r=rafiss

otan commented:
bors r=rafiss
